airflow/providers/microsoft/azure/hooks/data_factory.py

```diff
@@ -643,7 +642,7 @@ def wait_for_pipeline_run_status(
             "factory_name": factory_name,
             "resource_group_name": resource_group_name,
         }
-        pipeline_run_status = self.get_pipeline_run_status(**pipeline_run_info)
+        pipeline_run_status = self.get_pipeline_run_status(**pipeline_run_info)  # type: ignore
```

Comment:
This feels unnecessary. What is the error we’re trying to work around here?

Reply:
```
airflow/providers/microsoft/azure/hooks/data_factory.py:671: error: Argument 1 to "get_pipeline_run_status" of "AzureDataFactoryHook" has incompatible type "**Dict[str, Optional[str]]"; expected
"str"
            pipeline_run_status = self.get_pipeline_run_status(**pipeline_run_info)
```
If I enforce keywords arguments on the `get_pipeline_run_status()` function, mypy doesn't complain. Although, we could always explicitly pass a dictionary rather than using the `**kwargs` syntax.

Reply:
Oh I guess I could use a `TypedDict` here

Reply:
`pipeline_run_info: Dict[str, Any] = ` resolves the check as well but it's not _entirely_ a correct typing.